utide/_time_conversion.py

```diff
@@ -2,44 +2,79 @@
 Utility for allowing flexible time input.
 """
 
-import warnings
+import numpy as np
 
-from datetime import date, datetime
 
+# to be added to get 1 on 1st January of year 1 from unix epoch '1970-01-01'
+_DAY_TO_GREGORIAN_EPOCH = 719163
 
-try:
-    from datetime import timezone
+# milisecond in a day
+_MS_PER_DAY = 1000 * 86400
 
-    have_tz = True
-except ImportError:
-    have_tz = False
+# datenum types
+_DATENUM_TYPES = (np.float, np.int64, np.int32)
 
 
-def _normalize_time(t, epoch):
-    if epoch == "python":
-        return t
-    if epoch == "matlab":
-        return t - 366
+def _date2num(date, epoch="1970-01-01 00:00:00.000"):
+    """
+    Numpy based date to datenum calculator.
+
+    `date` can be anything parsable by np.datetime64 - string, datetime, datetime64,
+    pandas datetime.
+
+    `epoch` is taken as the unix epoch.
```

Comment:
It doesn't need to be a kwarg then, does it?

Reply:
Updated the function call. 

BTW, according to the documentation, the atleast_1d does not change the dimension of higher order array. From the doc 

> Scalar inputs are converted to 1-dimensional arrays, whilst higher-dimensional inputs are preserved.

Yes, we can leave out the `epoch` keyword for this case. However if there is another change in the future, although unlikely, having this keyword might help. I propose to keep it, and it does not hurt as it is a "private" function.

I have updated the docstring.